### ISSUE TITLE
OneSignal notifications with "Additional Data" are failing

### ISSUE THREAD

**Issue by vincnetas** [mautic/mautic#6709]
What type of report is this: Bug report

## Description:

When sending push notification with OneSignal plugin, generated JSON body looks like this:

```
{
  "include_player_ids" : [ "*******" ],
  "contents" : {
    "en" : "Happy"
  },
  "headings" : {
    "en" : "Tesla"
  },
  "data" :  [{
    "label" : "Data",
    "value" : "Value"
  }] ,
  "subtitle" : {
    "en" : "Subtitle"
  },
  "ios_sound" : "default",
  "ios_badgeType" : null,
  "ios_badgeCount" : 0,
  "content_available" : false,
  "ios_attachments" : {
    "id_5bb604839338e" : null
  },
  "mutable_content" : false,
  "android_sound" : null,
  "small_icon" : null,
  "large_icon" : null,
  "big_picture" : null,
  "android_led_color" : "FF",
  "android_accent_color" : "FF",
  "android_group" : null,
  "android_visibility" : 0,
  "app_id" : "****"
}
```

Problem is with "data" field. According to one signal documentation (https://documentation.onesignal.com/reference#section-attachments) data field must be a map. Mautic generates array :

```
"data" :  [{
    "label" : "Data",
    "value" : "Value"
  }]
```

Should be :

```
"data" :  {
    "Data" : "Value"
  }

``` 

One signals gives following response to invalid request :

```
{
    "errors": [
        "Data must be a valid JSON object and cannot be a string, a number, true, or false. See bit.ly/2iBky3H for an example of a JSON object. You can use jsonlint.com as a JSON validator."
    ]
}
```

| Q   | A
| --- | ---
| Mautic version |2.14.1 |
| PHP version | 7.1.16|

### Steps to reproduce:
1. Create Mobile notification.
2. Under "Additional data" tab add any custom values.
3. Send push notification using one signal api

 
### Log errors: 

```
{
    "errors": [
        "Data must be a valid JSON object and cannot be a string, a number, true, or false. See bit.ly/2iBky3H for an example of a JSON object. You can use jsonlint.com as a JSON validator."
    ]
}
```

**Comment by vincnetas** [comment]
For now implemented local fix in "app/bundles/NotificationBundle/Api/OneSignalApi.php"

```
                case 'additional_data':
                    $result = [];

                    foreach($value['list'] as $item) {
                        $result[$item["label"]] = $item["value"];
                    }

                    $data['data'] = $result;
                    break;
```

**Comment by Woeler** [comment]
@vincnetas Please apply this pr and confirm it fixes your issue https://github.com/mautic/mautic/pull/6711

**Comment by vincnetas** [comment]
Checked without additional and push notifications not working. Will need to investigate further to get request that's being sent by mautic to onesignal.

**Comment by Woeler** [comment]
Did you use the pr code? If so can you let me know what issue it has for you?

**Comment by vincnetas** [comment]
Ok, i got some time to check it so it's working with "additional data". Reason why it was not woring was because "iOS Message subtitle" was not entered in iOS settings. But this is not related to current fix, so i say that it's woking.

**Comment by Woeler** [comment]
So are you saying it worked all along, or that you applied the PR and that it worked with the PR applied? Because if it works without the PR I can close that one.

**Comment by Woeler** [comment]
I'm still waiting for feedback if this was working before or after the PR was applied. In the mean time I'll take it off the timeline.

**Comment by npracht** [comment]
@Woeler according to last comments, what should we do with that issue ?

**Comment by Woeler** [comment]
I can push this change, the thing is. I have no way to test if this works, because I do not use onesignal

**Comment by npracht** [comment]
@vincnetas would you be able to test a PR ?

**Comment by npracht** [comment]
@vincnetas last call!

**Comment by npracht** [comment]
@vincnetas can you please test #8605 ?

**Comment by dennisameling** [comment]
Fixed in https://github.com/mautic/mautic/pull/8605 which will be released in Mautic 3.2.5 on January 25 👍🏼

**Comment by namratasaun** [comment]
for me mobile notification, not working. it just fails sending without any logs. any leads as to what I might be missing?
